ee/server/service/mdm.go

```diff
@@ -246,3 +246,76 @@ func (svc *Service) GetMDMAppleBootstrapPackageSummary(ctx context.Context, team
 
 	return summary, nil
 }
+
+func (svc *Service) MDMAppleCreateEULA(ctx context.Context, name string, f io.ReadSeeker) error {
+	if err := svc.authz.Authorize(ctx, &fleet.MDMAppleEULA{}, fleet.ActionRead); err != nil {
+		return err
+	}
+
+	if err := file.CheckPDF(f); err != nil {
+		if errors.Is(err, file.ErrInvalidType) {
+			return &fleet.BadRequestError{
+				Message:     err.Error(),
+				InternalErr: err,
+			}
+		}
+
+		return ctxerr.Wrap(ctx, err, "checking pdf")
+	}
+
+	// ensure we read the file from the start
+	_, err := f.Seek(0, io.SeekStart)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "seeking start of PDF file")
+	}
+
+	bytes, err := io.ReadAll(f)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "reading EULA bytes")
+	}
+
+	eula := &fleet.MDMAppleEULA{
+		Name:  name,
+		Token: uuid.New().String(),
+		Bytes: bytes,
+	}
+
+	if err := svc.ds.MDMAppleInsertEULA(ctx, eula); err != nil {
+		return ctxerr.Wrap(ctx, err, "inserting EULA")
+	}
+
+	return nil
+}
+
+func (svc *Service) MDMAppleGetEULABytes(ctx context.Context, token string) (*fleet.MDMAppleEULA, error) {
+	// skipauth: this resource is authorized using the token provided in the
+	// request.
+	svc.authz.SkipAuthorization(ctx)
+
+	return svc.ds.MDMAppleGetEULABytes(ctx, token)
+}
+
+func (svc *Service) MDMAppleDeleteEULA(ctx context.Context, token string) error {
+	if err := svc.authz.Authorize(ctx, &fleet.MDMAppleEULA{}, fleet.ActionRead); err != nil {
```

Comment:
Should delete be `ActionWrite`?

Reply:
yes! and this is not the first time you catch this in a PR of mine. Thanks! I'll be more careful